app/src/main/java/com/klk/dailyjournal/SecondActivity.kt

```diff
@@ -0,0 +1,227 @@
+
+
+    val TAG = "xyz"
+    private val TAKE_PHOTO_PERMISSION_REQUEST_CODE = 1
+
+
+    fun saveNote(view: View){
+        val gratefulnessEdit = findViewById<TextView>(R.id.gratefulness_edit)
+        val todayEdit = findViewById<TextView>(R.id.today_edit)
+        val myNoteEdit = findViewById<TextView>(R.id.my_note_edit)
+
+        val path  = if (pathPhoto==null)  photo?.path else pathPhoto
+        repo.insert(NoteEntity(
+            0,
+            SimpleDateFormat("EEEE", Locale.ENGLISH).
+            format(getDate()),
+            "${getDayAsString()} ${getMonthName()}",
+            "face_temporary.png",
+            gratefulnessEdit.text.toString(),
+            todayEdit.text.toString(),
+            myNoteEdit.text.toString(),
+            path, null))
+
+            //save
+    }
+
+    fun pickImageFromGallery(view: View){
+        val intent = Intent()
+        intent.type = "image/*"
+        intent.action = Intent.ACTION_GET_CONTENT
+        intent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, false)
+        singleImageResultLauncher.launch(Intent.createChooser(intent, "Select Picture"))
+    }
+
+    val singleImageResultLauncher =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == Activity.RESULT_OK) {
+                val data: Intent? = result.data
+                val selectedImageUri: Uri? = data?.data
+                if (null != selectedImageUri) {
+                    findViewById<ImageView>(R.id.take_photo_img).setImageURI(selectedImageUri)
+                    photo = null
+                    pathPhoto = getPathFromURI(selectedImageUri)
+
+                }
+            }
+        }
+
+    private fun getPathFromURI(uri: Uri?): String {
+        var path = ""
+        if (contentResolver != null) {
+            val cursor = contentResolver.query(uri!!, null, null, null, null)
+            if (cursor != null) {
+                cursor.moveToFirst()
+                val idx = cursor.getColumnIndex(MediaStore.Images.ImageColumns.DISPLAY_NAME)
+                path = cursor.getString(idx)
+                cursor.close()
+            }
+        }
+        return path
+    }
+
+
+    private fun getDate(): Date{
+        return calendar.time
+    }
+
+    private fun getMonthName():String{
+        val month_date = SimpleDateFormat("MMMM")
+        return month_date.format(calendar.time)
+    }
+
+    private fun  getDayAsString(): String{
+        val day = calendar.get(Calendar.DAY_OF_MONTH)
+        var dayStr = ""
+        if(day.toString().endsWith('1'))
+            dayStr = day.toString() +"st"
+        else if(day.toString().endsWith('2'))
+            dayStr = day.toString() +"nd"
+        else if(day.toString().endsWith('3'))
+            dayStr = day.toString() +"rd"
+        else
+            dayStr = day.toString() + "th"
+
+        return dayStr
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
```

Comment:
this is just a detailed but I feel like functions like 'onCrete' should be kept on top of the class